packages/components/src/public/QueryModel/TabbedGridPanel.tsx

```diff
@@ -0,0 +1,130 @@
+    model: QueryModel;
+    onSelect: (id: string) => void;
+    pullRight: boolean;
+}
+
+const GridTab: FC<GridTabProps> = memo(({ isActive, model, onSelect, pullRight }) => {
+    const { id, queryInfo, title } = model;
+    const className = classNames({
+        active: isActive,
+        'pull-right': pullRight,
+    });
+    const onClick = useCallback(() => onSelect(id), [id, onSelect]);
+
+    return (
+        <li className={className}>
+            <a onClick={onClick}>{title || queryInfo.queryLabel || queryInfo.name}</a>
+        </li>
+    );
+});
+
+interface TabbedGridPanelProps<T = {}> extends GridPanelProps<T> {
+    /**
+     * The id of the model you want to render. Required if you are using onTabSelect. If passed when not using
+     * onTabSelect it will be used as the initial active tab.
+     */
+    activeModelId?: string;
+    /**
+     * Defaults to true. Determines if we render the TabbedGridPanel as a Bootstrap panel.
+     */
+    asPanel?: boolean;
+    /**
+     * Optional, if used the TabbedGridPanel will act as a controlled component, requiring you to always pass the
+     * activeModelId. If not passed the TabbedGridPanel will maintain the activeModelId state internally.
+     * @param string
+     */
+    onTabSelect?: (modelId: string) => void;
+    /**
+     * The model IDs you want to render as tabs that are pulled to the right side of the tabs.
+     */
+    rightTabs?: string[];
+    /**
+     * Array of model ids representing the order you want the tabs in. This component will only render Tabs and
+     * GridPanels for Query Models in the TabOrder array.
+     */
+    tabOrder: string[];
+    /**
+     * The title to render, only used if asPanel is true.
+     */
+    title?: string;
+}
+
+export const TabbedGridPanel: FC<TabbedGridPanelProps & InjectedQueryModels> = memo(props => {
+    const {
+        activeModelId,
+        actions,
+        asPanel = true,
+        onTabSelect,
+        queryModels,
+        rightTabs = [],
+        title,
+        tabOrder,
+        ...rest
+    } = props;
+    const [internalActiveId, setInternalActiveId] = useState<string>(activeModelId ?? tabOrder[0]);
+    const onSelect = useCallback(
+        (modelId: string) => {
+            if (onTabSelect !== undefined) {
+                onTabSelect(modelId);
+            } else {
+                setInternalActiveId(modelId);
+            }
+        },
+        [onTabSelect]
+    );
+    // If the component is passed onTabSelect we will only honor the activeModelId passed to this component.
+    const activeId = onTabSelect === undefined ? internalActiveId : activeModelId;
+    const activeModel = queryModels[activeId];
+
+    return (
+        <div className={classNames('tabbed-grid-panel', { panel: asPanel, 'panel-default': asPanel })}>
+            {title !== undefined && asPanel && (
+                <div className="tabbed-grid-panel__title panel-heading">
+                    <span>{title}</span>
```

Comment:
just curious, was there a specific reason for needing to put the `title` in a `span` here?

Reply:
Easier to query for when testing.

Reply:
Actually, no, I think I was maybe doing something else that made sense to have this span, but changed plans. I did something else recently that had a title and an icon or something and I put both in a span so tests could query and assert for exact text instead of containing text, or text plus a space, but that is clearly not this case.